src/lib/y2storage/blk_device.rb

```diff
@@ -395,7 +426,15 @@ def component_of
     # @return [Array<String>]
     def component_of_names
       # So far, all the possible elements on the array respond to #name
-      component_of.map(&:name)
+      component_of.map do |dev|
+        if dev.respond_to?(:name)
+          dev.name
+        elsif dev.respond_to?(:uuid)
+          dev.uuid
+        else
+          raise "Unknown tyoe if device #{dev.inspect}"
```

Comment:
NP: The error message looks wrong.

Reply:
I don't know if displaying an UUID will help the users to know that they are removing the cset from a given bcache device. Something including the bcache name would probably be more informative.

On the other hand, raising an exception here may be a little bit radical. :wink: